tests/transforms/test_unbroadcast_expand.py

```diff
@@ -0,0 +1,203 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import numpy as np
+import pennylane as qml
+from pennylane import numpy as pnp
+
+
+class RX_batched(qml.RX):
+    """A version of qml.RX that detects batching."""
+
+    ndim_params = (0,)
+    compute_decomposition = staticmethod(lambda theta, wires=None: qml.RX(theta, wires=wires))
```

Comment:
I suppose this implies we still need to go back and update all the ops? 😬

Reply:
Ah, yes, this will happen in the updated version of #2535 :)
Note that only the `ndim_params` will be necessary. The `compute_decomposition` addition is just required to make `RX_batched` executable on a device (via decomposition).